pocs/camera/fli.py

```diff
@@ -0,0 +1,236 @@
+import time
+import os
+from threading import Event
+from threading import Timer
+from threading import Lock
+
+import numpy as np
+
+from astropy import units as u
+from astropy.io import fits
+
+from pocs.camera.camera import AbstractCamera
+from pocs.camera import libfli
+from pocs.utils.images import fits as fits_utils
+
+
+class Camera(AbstractCamera):
+
+    def __init__(self,
+                 name='FLI Camera',
+                 set_point=None,
+                 filter_type='M',
+                 *args, **kwargs):
+        kwargs['readout_time'] = 1.0
+        kwargs['file_extension'] = 'fits'
+        super().__init__(name, *args, **kwargs)
+
+        # Create a Lock that will be used to prevent overlapping exposures.
+        self._exposure_lock = Lock()
+
+        # Create an instance of the FLI Driver interface
+        self._FLIDriver = libfli.FLIDriver()
+        self.connect()
+
+        self.filter_type = filter_type
+
+        if self.is_connected:
+            if set_point is None:
+                # FLI *always* thermoregulates, all you can do is set a high set point
+                set_point = 25 * u.Celsius
+            self.CCD_set_point = set_point
+            self.logger.info('{} initialised'.format(self))
+
+# Properties
+
+    @AbstractCamera.uid.getter
+    def uid(self):
+        # Unlike Canon DSLRs 1st 6 characters of serial number is *not* a unique identifier.
+        # Need to use the whole thing.
+        return self._serial_number
+
+    @property
+    def CCD_temp(self):
```

Comment:
Should be avoiding the capital letters in methods and properties.

Reply:
Torn between two conventions :-) Should be lower case from Python conventions point of view, should be upper case as 'CCD' is an acronym. I'll lower them if you prefer.